DCHdigi/src/DCHdigi_v02.cpp

```diff
@@ -0,0 +1,375 @@
+
+            // Calculate the number of clusters for each particle in the cell
+            for (const auto& [object_id, particle_info] : cluster_info_map) {
+                // Get number of clusters per length from delphes
+                // Output from delphes function is in 1/m, so to convert to 1/mm we need to scale accordingly
+                double nclusters_per_mm = m_delphesTrkUtil.Nclusters(particle_info.beta_gamma, m_GasType.value()) / 1000.0;
+                double nclusters_mean = nclusters_per_mm * particle_info.path_length_mm;
+
+                total_nclusters += sample_zero_truncated_poisson(nclusters_mean, random_engine);
+            }
+
+
+
+            /////////////////////////////
+            // SAVING THE SENSEWIREHIT //
+            /////////////////////////////
+            
+            // Use the first hit in the train for the time, position, and distance to wire
+            const auto& first_hit = hit_train.front();
+            auto sense_wire_hit = output.create();
+            sense_wire_hit.setCellID(cellID);
+            sense_wire_hit.setType(0);
+            sense_wire_hit.setQuality(0);
+            sense_wire_hit.setTime(first_hit.arrival_time_ns);
+            sense_wire_hit.setEDep(edep_sum_GeV);
+            sense_wire_hit.setEDepError(0.0);
+            sense_wire_hit.setPosition(first_hit.position_mm);
+            sense_wire_hit.setPositionAlongWireError(m_z_resolution_mm);
+            sense_wire_hit.setWireAzimuthalAngle(WireAzimuthalAngle);
+            sense_wire_hit.setWireStereoAngle(WireStereoAngle);
+            sense_wire_hit.setDistanceToWire(first_hit.distance_to_wire_mm);
+            sense_wire_hit.setDistanceToWireError(m_xy_resolution_mm);
+
+            // Clusters are added to the SenseWireHit as vector member containing the number of electrons in each cluster
+            // The length of this vector is the total number of clusters in the cell
+            // For now, we do not calculate the size of each cluster, so we just fill it with a dummy value
+            for (unsigned int i = 0; i < total_nclusters; ++i) {
+                sense_wire_hit.addToNElectrons(-1);
+            }
+
+            /////////////////////
+            // SAVING THE LINK //
+            /////////////////////
+
+            auto link = links.create();
+            link.setFrom(sense_wire_hit);
+            link.setTo(*(first_hit.simhit));
+
+        } // end of loop over hit_train_vector
+        
+    } // end of loop over the cells
+
+    return std::make_tuple(std::move(output), std::move(links));
+
+}
+
+
+double DCHdigi_v02::get_drift_time(double distance_to_wire_mm) const {
+    // Calculate the drift time based on the distance to the wire
+    // This is a preliminary implementation that needs to be updated with a realistic model
+    // For now, we use a simple linear model with a constant drift velocity
+
+    double drift_velocity_um_per_ns = 25.0;
```

Comment:
I'd propose to make it a Gaudi option and propagate it as an argument

Reply:
Yes, my point is that since we provide the possibility to change the gas type for dN/dx, we should also provide flexibility for the drift velocity. Maybe we could tight it to the gas type (no need for an additional option then), with a switch and print a warning in case something else than He-Isobutane is chosen